src/sentry/search/django/backend.py

```diff
@@ -8,286 +8,445 @@
-                pass
-            else:
-                teams = member.get_teams()
-
-            queryset = queryset.filter(
-                Q(assignee_set__user=assigned_to, assignee_set__project=project) |
-                Q(assignee_set__team__in=teams)
-            )
-        elif unassigned in (True, False):
-            queryset = queryset.filter(
-                assignee_set__isnull=unassigned,
-            )
+class Condition(object):
+    """\
+    Adds a single filter to a ``QuerySet`` object. Used with
+    ``QuerySetBuilder``.
+    """
+
+    def apply(self, queryset, name, parameters):
+        raise NotImplementedError
+
+
+class CallbackCondition(Condition):
+    def __init__(self, callback, destructive=True):
+        self.callback = callback
+        self.destructive = destructive
+
+    def apply(self, queryset, name, parameters):
+        accessor = parameters.pop if self.destructive else parameters.__getitem__
+        return self.callback(queryset, accessor(name))
+
+
+class ScalarCondition(Condition):
+    """\
+    Adds a scalar filter (less than or greater than are supported) to a
+    ``QuerySet`` object. Whether or not the filter is inclusive is defined by
+    the '{parameter_name}_inclusive' parameter.
+    """
+
+    def __init__(self, field, operator, destructive=True):
+        assert operator in ['lt', 'gt']
+        self.field = field
+        self.operator = operator
+        self.destructive = destructive
+
+    def apply(self, queryset, name, parameters):
+        accessor = parameters.pop if self.destructive else parameters.__getitem__
+        inclusive = accessor('{}_inclusive'.format(name))
+        return queryset.filter(**{
+            '{}__{}{}'.format(
+                self.field,
+                self.operator,
+                'e' if inclusive else ''
+            ): accessor(name)
+        })
+
+
+def get_sql_table(model):
+    return '{}'.format(model._meta.db_table)
+
+
+def get_sql_column(model, field):
+    "Convert a model class and field name to it's (unquoted!) SQL column representation."
+    return '{}.{}'.format(*[
+        get_sql_table(model),
+        model._meta.get_field_by_name(field)[0].column,
+    ])
+
+
+sort_strategies = {
+    # sort_by -> Tuple[
+    #   Paginator,
+    #   String: QuerySet order_by parameter
+    # ]
+    'priority': (Paginator, '-score'),
+    'date': (DateTimePaginator, '-last_seen'),
+    'new': (DateTimePaginator, '-first_seen'),
+    'freq': (Paginator, '-times_seen'),
+}
+
+
+def get_priority_sort_expression(model):
+    engine = get_db_engine(router.db_for_read(model))
+    table = get_sql_table(model)
+    if 'postgres' in engine:
+        return 'log({table}.times_seen) * 600 + {table}.last_seen::abstime::int'.format(table=table)
+    else:
+        # TODO: This should be improved on other databases where possible.
+        # (This doesn't work on some databases: SQLite for example doesn't
+        # have a built-in logarithm function.)
```

Comment:
This can probably be done on the other database but I didn't really look into it, and really just wanted tests to pass.